### DISCUSSION TITLE
Transparency issue when drawing

### DISCUSSION THREAD

**Discussion by miles-tbe** [pelicanmapping/osgearth#2022] [Support Q&A]
Hi, 
I have a large volume I am drawing and from certain angles I am seeing an effect I do not wish to see when it comes to the transparency of it. Where some of the volume is darker and some of it is lighter, especially in this disc-like shapes on the right.
![osgEarth Volume](https://user-images.githubusercontent.com/93147818/159305244-3114e72d-6dbf-4f82-a884-c559062fa975.png)
 
Compared to how I would expect/want to see the same volume, where it is all the same uniform shade and transparency:
![Standard Volume](https://user-images.githubusercontent.com/93147818/159305403-1b931e3e-5ef8-4cdf-84e0-58cbe3071696.png)

All of my transparency options are set in a GeoTransform object. And the settings I have are as follows: 
```cpp
Volume->getOrCreateStateSet()->setMode (GL_BLEND, 1);
Volume->getOrCreateStateSet()->setRenderingHint(osg::StateSet::TRANSPARENT_BIN);
Volume->getOrCreateStateSet()->setAttributeAndModes (new osg::BlendFunc (GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA));
```

I was looking into why this might be and stumbled upon these 2 options as a potential solution:
```cpp
Volume->getOrCreateStateSet ()->setMode (GL_DEPTH_TEST, 1);   
Volume->getOrCreateStateSet()->setAttributeAndModes (new osg::Depth (osg::Depth::LEQUAL, 1, 0, true));
```

Is this the correct path to go down? If so, what would you suggest I do? I tried this piece of code and while it got rid of the disc-like pieces on the curved part of the volume, it also made it opaque. Or if it is not, do you what is causing this behavior and how to fix it?

Thanks,
Miles

**Comment by RhythmOfTension** [comment]
Look into `AnnotationUtils::installTwoPassAlpha` method. It's worked for me in case of view of volume, however i have another issue with rendering other objects inside my sphere... elevation and image layers are fine, but objects from featuremodel layers are not visible if sphere is closer to the camera. I guess i should do something with RenderSymbol of model layer objects? In your case seems like they are rendering just fine.

**Reply by miles-tbe** [reply]
I did not find installTwoPassAlpha to solve my issue in particular but thank you for the suggestion.
